adapters/adorsys-adapter/src/main/java/de/adorsys/xs2a/adapter/adorsys/AdorsysIntegServiceProvider.java

```diff
@@ -45,35 +41,25 @@ public PaymentInitiationService getPaymentInitiationService(Aspsp aspsp,
                                                                 LinksRewriter linksRewriter) {
         return new BasePaymentInitiationService(aspsp,
                                                 httpClientFactory.getHttpClient(getAdapterId()),
-                                                keyStore != null ? getInterceptor(keyStore) : null,
+                                                Arrays.asList(oauthHeaderInterceptor,
+                                                    getAdorsysSigningHeadersInterceptor(keyStore)),
                                                 linksRewriter);
     }
 
-    //todo: https://jira.adorsys.de/browse/XS2AAD-706
-    Interceptor getInterceptor(Pkcs12KeyStore keyStore) {
-        if (requestSigningEnabled) {
-            RequestSigningInterceptor requestSigningInterceptor = new RequestSigningInterceptor(keyStore);
-            return requestBuilder -> {
-                oauthHeaderInterceptor.preHandle(requestBuilder);
-                requestBuilder.header(RequestHeaders.DATE,
-                                      DateTimeFormatter.RFC_1123_DATE_TIME.format(ZonedDateTime.now()));
-                requestSigningInterceptor.preHandle(requestBuilder);
-                String certificate = requestBuilder.headers().get(RequestHeaders.TPP_SIGNATURE_CERTIFICATE);
-                requestBuilder.header(RequestHeaders.TPP_SIGNATURE_CERTIFICATE,
-                                      "-----BEGIN CERTIFICATE-----" + certificate + "-----END CERTIFICATE-----");
-                return requestBuilder;
-            };
-        }
-        return oauthHeaderInterceptor;
+    private AdorsysSigningHeadersInterceptor getAdorsysSigningHeadersInterceptor(Pkcs12KeyStore keyStore) {
+        return nonNull(keyStore) ? new AdorsysSigningHeadersInterceptor(keyStore) : null;
     }
 
     @Override
     public AccountInformationService getAccountInformationService(Aspsp aspsp,
                                                                   HttpClientFactory httpClientFactory,
                                                                   Pkcs12KeyStore keyStore,
                                                                   LinksRewriter linksRewriter) {
-        return new AdorsysAccountInformationService(aspsp, httpClientFactory.getHttpClient(getAdapterId()),
-                                                    getInterceptor(keyStore), linksRewriter);
+        return new AdorsysAccountInformationService(aspsp,
+                                                    httpClientFactory.getHttpClient(getAdapterId()),
+                                                    Arrays.asList(oauthHeaderInterceptor,
+                                                        getAdorsysSigningHeadersInterceptor(keyStore)),
```

Comment:
I would suggest do not put null into the list

Reply:
refined